presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/Histogram.java

```diff
@@ -125,16 +126,18 @@ public static void input(Type type, HistogramState state, Block key, int positio
 
     public static void combine(HistogramState state, HistogramState otherState)
     {
-        if (state.get() != null && otherState.get() != null) {
-            TypedHistogram typedHistogram = state.get();
-            long startSize = typedHistogram.getEstimatedSize();
-            typedHistogram.addAll(otherState.get());
-            state.addMemoryUsage(typedHistogram.getEstimatedSize() - startSize);
-        }
-        else if (state.get() == null && otherState.get() != null) {
-            TypedHistogram otherTypedHistogram = otherState.get();
-            state.set(otherTypedHistogram);
-        }
+        // NOTE: state = current merged state; otherState = scratchState (new data to be added)
+
+        // since we have a single instance of a histogram, it will never be non-null. Even asking for a histogram that does not exist returns the histogram object. It doesn't
+        // "exist" until we call add() on it,though
```

Comment:
nit: What about the following ? (Is my understanding correct ? :)
```
A histogram object will be returned even the group is empty. 
In that case, the histogram object will represent an empty histogram until we call add() on it.
```

Reply:
yes, an object is return, but there is no histogram for that group until add is called. think like a map. there is no value for key k1 until i do map.put(j1,v1)

it's just implicit in context what the groupId is